docs/tests.md

```diff
@@ -0,0 +1,439 @@
+* Parallel: Yes
+* ExecTime: ~5s on AWS
+* Notes: AWS only
+* Recommendation:
+  * Might be removed. Such functionality might be covered with webhook tests in MAPO
+
+This test excersising validation webhook behaviour for `metadataServiceOptions.authentication` field. Checks that machineset can not be created
+if `metadataServiceOptions.authentication` is not equal `Required` or `Optional`
+
+#### MetadataServiceOptions should enforce auth on metadata service if metadataServiceOptions.authentication set to Required
+#### MetadataServiceOptions should allow unauthorized requests to metadata service if metadataServiceOptions.authentication is Optional
+* Parallel: Yes (each)
+* ExecTime: 4-5m on AWS (each)
+* Notes: AWS only
+
+Each of these tests creates a machineset with a respective value (`Required` or `Optional`) in `metadataServiceOptions.authentication`.
+Then, after a machine from the created machineset hits "Running" phase,  pod with simple curl command starts on the respective node then.
+Test waits for the curl output and checks if it contains expected response (401 for required case, 200 for optional).
+
+### Managed Cluster (common machinesets tests)
+Every single test in this section creates a manineset with 2 machines, because of that, this tests
+takes a significant time (>5 m).
+
+All of the test listed below shares common `BeforeEach` and `AfterEach` blocks.
+In `BeforeEach` machineset creation and wait happen (this machineset has 2 desired replicas configured, so 2 machines will be created for each test here).
+In `AfterEach` created machineset is being deleted.
+
+#### Managed cluster should drain node before removing machine resource
+* Parallel: Yes
+* ExecTime: >12m on AWS
+* Recommendations:
+  * Update resource creation helper functions, and add parameters to specify workload names, replicas count, selectors, etc. Now it's almost impossible to say what is going on without looking into these functions.
+  * Specify selectors more strictly for the created workloads, now it's theoretically possible that created workload will be scheduled somewhere else and we could get false positive results.
+  * Add more meaningful comments inside the test code, now it's not easy to figure out what is going on there.
+  * Fix logging here, it is quite noisy, and do flood reports.
+
+This test checks nodes are properly draining during machine deletion and this process also respects PDBs (pod disruption budgets).
+During this test - a replication controller creating with 20 desired replicas, then PDB creating for this workload.
+After the replication controller is ready, machineset marking for deletion. Along with the deletion process test checking that pods amount on the node 
+is going down to 1 and RC has at most one non-ready replica.
+
+Test considered failed in case if more than 1 replica within RC detected as not ready during machineset deletion. That would mean that PDB constraints were violated.
```

Comment:
Where do the pods go to when the machineset is removed?  Do we really need to run 20 pods?